plugins/module_utils/foreman_helper.py

```diff
@@ -201,6 +201,16 @@ def find_resource(self, resource, search, name=None, params=None, failsafe=False
                 result = self.show_resource(resource, result['id'], params=params)
         return result
 
+    def find_katello_resource_by_name(self, resource, organization, name, **kwargs):
+        search = 'name="{}"'.format(name)
+        kwargs['name'] = name
+        if 'params' in kwargs:
+            kwargs['params'].update({'organization_id': organization['id']})
+        else:
+            kwargs['params'] = {'organization_id': organization['id']}
+
+        return self.find_resource(resource, search, **kwargs)
+
```

Comment:
Not sure, i like this. It's easy to keep a scope variable around in the module and pass that to the params arg.
Also there are cases, where you do not need / want to pass the organization_id (for example when obtaining subscriptions for an activation_key [@evgeni might disagree on that example] or reading repositories from a product), because they are already scoped more specifically.

Reply:
I won't disagree on the "do not need" (I might on "do not want", but that's not for discussion here :stuck_out_tongue:)

Having scoped resources is not a Katello-only thing, it's just that Katello *requires* them to be scoped, while Foreman does not. It is perfectly fine in a plain Foreman to have two Orgs which both contain a hostgroup called "prod" and without scoping the search you will never find the right one if your user can see both.

So if anything, we should have `find_scoped_resource…` helpers, but then again just passing the right scope as we did in the other Katello modules that were migrated to Apypie is (IMHO) equally easy.

What we could do (in the future) is to teach `ForemanEntityApypieAnsibleModule` to gather the scope "automatically" if a `organization`/`location` param was passed to the module and apply that scope in `find_resource` *unless* the developer provided a different scope while calling the function. But I think we have bigger fish to fry until this optimization is worth it.